### PR TITLE
new example structure and better docs

### LINKED ISSUE #479: Streamline Examples
- [x] Investigate ordering of examples based on progression
- [x] Make example structures less cluttered
- [x] Group examples into bigger groups?
- [ ] EA and db EA could be merged into one to avoid confusion
- [x] Improve documentation of examples
- [x] Make them use the new Evolution Abstraction !!!

### PR DESCRIPTION
**Pull request by oliverweissl**
closes #479

### CODE REVIEW

**Comment by oliverweissl** [comment]
@JedMuff The most important part in this PR is whether the structure of examples make sense, also if the readme for the example collections are good to explain their use. If something is unclear pls lmk.

File: examples/1_simulator_basics/1b_custom_terrain/README.md
Comment (oliverweissl) [inline examples/1_simulator_basics/1b_custom_terrain/README.md:1]: should be 1a_

File: examples/3_experiment_foundations/3c_evaluate_multiple_isolated_robots/README.md
Comment (oliverweissl) [inline examples/3_experiment_foundations/3c_evaluate_multiple_isolated_robots/README.md:1]: This should be 3b

File: examples/3_experiment_foundations/3d_evaluate_multiple_interacting_robots/README.md
Comment (oliverweissl) [inline examples/3_experiment_foundations/3d_evaluate_multiple_interacting_robots/README.md:1]: make it a code block (example name)

Comment (oliverweissl) [reply examples/3_experiment_foundations/3d_evaluate_multiple_interacting_robots/README.md:1]: x-y displacement

File: examples/4_example_experiment_setups/4a_simple_ea_xor/main.py
Comment (oliverweissl) [inline examples/4_example_experiment_setups/4a_simple_ea_xor/main.py:1]: maybe make more comments in the selector classes

File: examples/4_example_experiment_setups/4c_robot_bodybrain_ea/README.md
Comment (oliverweissl) [inline examples/4_example_experiment_setups/4c_robot_bodybrain_ea/README.md:1]: should be 3c

**Review by JedMuff — CHANGES_REQUESTED** [review]
Overall I like this structure and the comments made to the code to make it more like a tutorial.

**Comment by oliverweissl** [comment]
@JedMuff 
Thanks for the feedback! I adjusted,.

**Review by JedMuff — APPROVED** [review]

**Pull request merged**